### ISSUE TITLE
[Development] Documentation for local debugging

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/pocket#183]
## Objective

Document how to use the `dlv` debugger in LocalNet.

## Origin Document

The `LocalNet` CLI and logging are useful, but we need to be able to step through the code in unit tests and in a dev environment

## Goals

- [ ] Anyone who spins up a V1 LocalNet should be able to easily use the debugger, regardless of whether they have experience with it in the past
- [ ] Support `vscode` and `Goland` as a first iteration

## Deliverables

- [ ] Documentation on how to use the `dlv` debugger using `vscode`
- [ ] Documentation on how to use the `dlv` debugger using `Goland`
- [ ] A gif or video with an example of how the debugger works
- [ ] Modifications to the supporting tooling (config files, docker-compose files, etc if necessary)

## Non-goals / Non-deliverables

- Modifying the core protocol implementation
- Enabling debugging in DevNet
- Introducing any new telemetry to the codebase

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @Olshansk 
**Co-Owners**: @andrewnguyen22

**Comment by Olshansk** [comment]
Copy pasting an idea from @bryanchriswhite to be documented in the docs in the future.

<img width="873" alt="Screenshot 2023-03-06 at 3 25 42 PM" src="https://user-images.githubusercontent.com/1892194/223279651-edc6cc5d-35a2-4afb-ae76-46da9585f46a.png">

```bash
If anyone else finds themselves debugging a  flickering / flaky test, may I recommend this guide which blew my mind: 🤯 http://choly.ca/post/debugging-go-with-rr/

TLDR; `go test -c ... + rr + dlv`:
1. go test -c ... builds a test binary which can be run by another program or script
2. rr runs in a loop (with the test) until produces a failure once. 
3. The captured execution trace can then be replayed deterministically and debugged with dlv. 🤙 
```

**Comment by Olshansk** [comment]
@red-0ne Are you down to take on this ticket?

**Comment by red-0ne** [comment]
Yes, dlv in unit tests are already working and helpful, would love to document it and take a look at the LocalNet scenario :+1:
